### PR TITLE
[NVIDIA] Fix convolution fusion

### PR DESCRIPTION
**Pull request by nkogteva**
- *Fix case when `Add` has `FusedConvolution` on both inputs which lead to hangs in mobilenet\* topologies*
- *Add transformation unit tests*

### CODE REVIEW

File: modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp [hunk 1]
```diff
@@ -121,21 +108,29 @@ struct FusedConvCallbacks {
...
+            if (get_node_id(fused_conv0) > get_node_id(fused_conv1)) {
+                return {fused_conv0, input1};
+            } else {
+                return {fused_conv1, input0};
+            }
         }
 
-        if (!fused_conv) {
-            return {nullptr, nullptr};
+        if (fused_conv0) {
```
Comment (vladimir-paramuzov) [inline modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp:126]: need to check node ids here as well to ensure that there's no dependency between fused_conv and input1

**Review by vladimir-paramuzov — CHANGES_REQUESTED** [review]

File: modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp [hunk 2]
```diff
@@ -273,11 +281,70 @@ bool fuse_convolution_backprop_data_with_add(Matcher &m) {
...
+    if (!node) {
+        return false;
+    }
+    if (node->input(0).get_partial_shape().is_dynamic() ||
+        node->input(1).get_partial_shape().is_dynamic()) {
+        return false;
+    }
+    const auto conv_shape = node->input(0).get_shape();
+    const auto bias_shape = node->input(1).get_shape();
+    const auto bias_rank = bias_shape.size();
```
[the remark this answers is not in the document]
Comment (nkogteva) [reply modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp:319]: Added

File: modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp [hunk 3]
```diff
@@ -273,11 +281,70 @@ bool fuse_convolution_backprop_data_with_add(Matcher &m) {
...
+    auto node = std::dynamic_pointer_cast<ov::op::v1::Add>(output.get_node_shared_ptr());
+    if (!node) {
+        return false;
+    }
+    const auto shape0 = node->input(0).get_partial_shape();
+    const auto shape1 = node->input(1).get_partial_shape();
+    if (shape0.is_dynamic() || shape1.is_dynamic()) {
+        return false;
+    }
+    return (shape0.to_shape() == shape1.to_shape());
```
[the remark this answers is not in the document]
Comment (nkogteva) [reply modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp]: Added check

File: modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp [hunk 4]
```diff
@@ -273,11 +281,70 @@ bool fuse_convolution_backprop_data_with_add(Matcher &m) {
...
+    }
+    const auto conv_shape = node->input(0).get_shape();
+    const auto bias_shape = node->input(1).get_shape();
+    const auto bias_rank = bias_shape.size();
+    if (bias_rank < conv_bias_rank_min || bias_rank > conv_bias_rank_max) {
+        return false;
+    }
+    const auto num_spatial_dims = conv_shape.size() - 2;
+    const auto nchw_channel_dim_reverse_offset = num_spatial_dims + 1;
+    if (bias_shape.at(bias_shape.size() - nchw_channel_dim_reverse_offset) !=
```
[the remark this answers is not in the document]
Comment (nkogteva) [reply modules/nvidia_plugin/src/transformer/fuse_conv_biasadd_activation.cpp]: Added check and test

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**